src/c++/perf_analyzer/request_rate_manager.cc

```diff
@@ -247,6 +247,23 @@ RequestRateManager::Infer(
   do {
     // Should wait till main thread signals execution start
     if (!execute_) {
+      if (on_sequence_model_) {
+        // Finish off all the ongoing sequences for graceful exit
+        for (size_t i = thread_config->id_; i < sequence_stat_.size();
```

Comment:
Why not just pick one thread to send the last request to all sequences and have others wait?

Reply:
That is a valid question, and something I considered. I either needed to wait and communicate through the threads, or wait and communicate through the sequences.  Using all of the threads to finish is faster, so I took that path.